totpauth-impl/src/main/resources/conf/authn/totp-authn-beans.xml

```diff
@@ -21,14 +21,15 @@
 
 	<bean id="TotpTokenValidator" class="net.kvak.shibboleth.totpauth.authn.impl.TotpTokenValidator" scope="prototype"
 		p:httpServletRequest-ref="shibboleth.HttpServletRequest" p:seedFetcher-ref="shibboleth.totp.seedfetcher" p:gAuth-ref="shibboleth.totp.gAuth" />
-        
-    <bean id="RegisterNewToken" class="net.kvak.shibboleth.totpauth.authn.impl.RegisterNewToken" scope="prototype"
+
+    <bean id="RegisterNewToken" class="net.kvak.shibboleth.totpauth.authn.impl.RegisterNewSeedSql" scope="prototype"
         p:httpServletRequest-ref="shibboleth.HttpServletRequest"
         p:gAuth-ref="shibboleth.totp.gAuth"
-        p:ldapTemplate-ref="ldapTemplate"
-        p:tokenCodeField-ref="shibboleth.authn.tokenCodeField">
-        <constructor-arg ref="shibboleth.authn.seedAttribute" />
-        <constructor-arg ref="shibboleth.authn.userAttribute" />
+        p:seedDataSource-ref="seedDataSource"
+        p:tokenCodeField-ref="shibboleth.authn.tokenCodeField"
+        p:seedDBTableName-ref="shibboleth.authn.seedDBTableName"
+        p:usernameColumnName-ref="shibboleth.authn.usernameColumnName"
+        p:seedColumnName-ref="shibboleth.authn.seedColumnName">
```

Comment:
Doesn't seem like the `RegisterNewSeedSql` class supports `backup_code` just yet, but shouldn't be a big deal for now.